~ $ cd Documents/Tutorials/django-tutorial/mysite
~/Documents/Tutorials/django-tutorial/mysite $ ./manage.py shell
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> Polls.objects.all()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'Polls' is not defined
>>> polls.objects.all()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'polls' is not defined
>>> Poll.objects.all()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'Poll' is not defined
>>> from polls.models import Poll, Choice
>>> Poll.objects.all()
[<Poll: What's up?>]
>>> p=Poll.objects.get(pk=1)
>>> print p
What's up?
>>> p.was_published_recently()
True
>>> p = Poll.objects.get(pk=1)
>>> p.choice_set.all()
[]
>>> p.choice_set.create(choice_text='Not much', vote=0)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/fields/related.py", line 533, in create
    return super(RelatedManager, self.db_manager(db)).create(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/manager.py", line 149, in create
    return self.get_query_set().create(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 412, in create
    obj = self.model(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/base.py", line 415, in __init__
    raise TypeError("'%s' is an invalid keyword argument for this function" % list(kwargs)[0])
TypeError: 'vote' is an invalid keyword argument for this function
>>> p.choice_set.create(choice_text='Not much', votes=0)
<Choice: Not much>
>>> p.choice_set.create(choice_text='The sky', votes=0\)
  File "<console>", line 1
    p.choice_set.create(choice_text='The sky', votes=0\)
                                                       ^
SyntaxError: unexpected character after line continuation character
>>> p.choice_set.create(choice_text='The sky', votes=0)
<Choice: The sky>
>>> p.choice_set.create(choice_text='Just hacking again', votes=0)
<Choice: Just hacking again>
>>> c.poo
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'c' is not defined
>>> c.poll
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'c' is not defined
>>> c = Choice.objects.get(pk=1)
>>> c.poll
<Poll: What's up?>
>>> c = Choice.objects.get(pk=2)
>>> c.poll
<Poll: What's up?>
>>> c = Choice.objects.get(id=2)
>>> c.poll
<Poll: What's up?>
>>> c = Choice.objects.get(id=3)
>>> c.poll
<Poll: What's up?>
>>> c = Choice.objects.get(id=4)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/manager.py", line 143, in get
    return self.get_query_set().get(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 401, in get
    (self.model._meta.object_name, kwargs))
DoesNotExist: Choice matching query does not exist. Lookup parameters were {'id': 4}
>>> Poll.objects.all()
[<Poll: What's up?>]
>>> Choice.objects.filter(poll__pub_date__year=current_year)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'current_year' is not defined
>>> print current_year
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'current_year' is not defined
>>> Choice.objects.filter(choice_text__startswith='J')
[<Choice: Just hacking again>]
>>> 
KeyboardInterrupt
>>> 

